woocommerce/woocommerce
# Revamp Featured page in WooCommerce Marketplace

### All Submissions:






### Changes proposed in this Pull Request:



Related to 11239-gh-Automattic/woocommerce.com.

We want to revamp the Marketplace page according to [Figma designs](seb2o1DExPxoh5d53ftJif-fi-812%3A13394). Together with the new design, the approach of treating data from WooCommerce.com API was changed as well. Instead of the current approach of sections/containers for products and banners, the new approach uses groups of products displayed one after another. 

Changes:
- Refactor current implementation of featured page blocks (the current version works on custom markup functions like "open container", put some product cards, "close container"; the update uses a simplified approach of showing groups of product cards instead)
- Add markup for showing ratings and review count for products
- Reuses current addon's product card for both search results, products by categories, and featured products
- CSS tweaks to make marketplace product card looks like on the design

### How to test the changes in this Pull Request:

#### Preparation
1. To test these changes properly you need an API for testing.
There're 2 options: 
- Local WCCOM (use 11456-gh-Automattic/woocommerce.com for this option) or
- Hardcoded JSON response from [this script](2a667-pb/)

For the 2nd option save the script in `index.php` file and launch PHP-Builtin server with the command like:
```sh
php -S host.docker.internal:9000
```

2. Disable using transient by changing the first line of the `render_featured` method in `includes/admin/class-wc-admin-addons.php` to: `$featured = false;`

3. Change the API URL used in `$raw_featured = wp_safe_remote_get(...` in the same method from `https://woocommerce.test/wp-json/wccom-extensions/2.0/featured` to `http://host.docker.internal:9000` (you may need to change the function used to `wp_remote_get` as well)

#### Testing
1. Open your testing WP site with WooCommerce on this branch
2. Open WP Admin > WooCommerce > Marketplace page, make sure that you are viewing the "Featured" section 
3. You should see something similar to the image below:
![image](https://user-images.githubusercontent.com/15897544/136460277-195c6afc-c655-4973-9adb-d8954264021b.png)

4. Try extensions search (e.g. via typing "memberships"), the page should look as expected


### Other information:




### Changelog entry

> Enhancement - Revamp the WooCommerce Marketplace page.

### FOR PR REVIEWER ONLY:
